### PR TITLE
MOTECH-2816 Refactor UI scheduler to be URI state driven

### PR DESCRIPTION
**Pull request by PJSosnowski**
If user click edit on specific task or click schedule job in scheduler module, its driven by url rather than by function in ng-click.

### CODE REVIEW

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by jredlarski** [comment]
test this please

**Comment by jredlarski** [comment]
@PJSosnowski please resolve conflicts

**Comment by jredlarski** [comment]
test this please

File: src/scheduler/scheduler.routes.js [hunk 1]
```diff
@@ -23,7 +23,7 @@
             }
         })
         .state('scheduler.createOrUpdateJob', {
-            url: "/createOrUpdateJob?action",
+            url: "/createOrUpdateJob?action?currJob",
```
Comment (jredlarski) [inline src/scheduler/scheduler.routes.js]: @PJSosnowski Change the second parameter to "jobName"

File: src/scheduler/create-or-update-job.controller.js [hunk 2]
```diff
@@ -165,6 +169,14 @@
         }
 
         LoadingModal.close();
+
+        function findJobByName (name, jobs) {
+            for (var i = 0; i < jobs.rows.length; i++) {
+                if (jobs.rows[i].name == name) {
```
Comment (jredlarski) [inline src/scheduler/create-or-update-job.controller.js]: Use `===`

**Pull request merged**